### PR TITLE
add autoscaler support

### PR DESCRIPTION
**Pull request by blublinsky**
## Why are these changes needed?

Enables autoscaling cluster creation by API server

## Related issue number

https://github.com/ray-project/kuberay/issues/1024

## Checks

- [x ] I've made sure the tests are passing. 
- Testing Strategy
   - [ x] Unit tests
   - [x ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by blublinsky** [comment]
@z103cb, @astefanutti, @tedhtchang, please take a look

cc @kevin85421

File: proto/cluster.proto [hunk 1]
```diff
@@ -144,6 +144,30 @@ message EnvironmentVariables {
   map<string, EnvValueFrom> valuesFrom = 2;
 }
 
+message AutoscalerOptions {
+  // IdleTimeoutSeconds is the number of seconds to wait before scaling down a worker pod which is not using Ray resources.
+	// Defaults to 60 (one minute).
+	int32 idleTimeoutSeconds = 1;
```
Comment (z103cb) [inline proto/cluster.proto:150]: Is this field required? If yes, let's adnotated as such.

Comment (blublinsky) [reply proto/cluster.proto:150]: nop, none of the fields are

File: proto/cluster.proto [hunk 2]
```diff
@@ -197,6 +221,11 @@ message ClusterSpec {
   HeadGroupSpec head_group_spec = 1 [(google.api.field_behavior) = REQUIRED];
   // Optional. The worker group configurations
   repeated WorkerGroupSpec worker_group_spec = 2;
+  // EnableInTreeAutoscaling indicates whether operator should create in tree autoscaling configs
+	bool enableInTreeAutoscaling = 3;
```
Comment (z103cb) [inline proto/cluster.proto:225]: Is the field required?

**Comment by z103cb** [comment]
Can you add an e2e test for this PR ?

File: proto/cluster.proto (hunk 2 above)
Comment (blublinsky) [reply proto/cluster.proto:225]: No, the cluster have to have head nodes, workers are optional

**Comment by blublinsky** [comment]
@z103cb This PR has unit test. E2e tests will be added as a separate PR. There are several that need to be added

File: apiserver/Autoscaling.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
+# Creating Autoscaling clusters using API server
+
+One of the fundamental features of Ray is autoscaling. This [document](https://docs.ray.io/en/latest/cluster/kubernetes/user-guides/configuring-autoscaling.html) describes how to set up autoscaling using Ray operator. Here we will describe how to set it up using API server.
+
+## Deploy KubeRay operator and API server
+
+Reffer to [readme](README.md) for setting up KubRay operator and API server.
```
Comment (tedhtchang) [inline apiserver/Autoscaling.md:7]:
```suggestion
Refer to [readme](README.md) for setting up KubRay operator and API server.
```

File: apiserver/Autoscaling.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+```shell
+make operator-image cluster load-operator-image deploy-operator
+```
+
```
Comment (tedhtchang) [inline apiserver/Autoscaling.md:12]:
```suggestion
Alternative, you could build and deploy the Operator and API server from local repo for development purpose.
```shell
make operator-image cluster load-operator-image deploy-operator docker-image load-image deploy
```

```

File: apiserver/Autoscaling.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+Additionally install this [configmap](test/cluster/cluster/detachedactor.yaml) containing code that we will use for testing
```
Comment (tedhtchang) [inline apiserver/Autoscaling.md:13]:
```suggestion
Additionally install this [configmap](test/cluster/cluster/detachedactor.yaml) containing code that we will use for testing
```shell
kubectl apply -f test/cluster/cluster/detachedactor.yaml
```
```

**Review by tedhtchang — APPROVED** [review]
Tested using the instruction with additional steps to make it easier for users.
Thanks
/LGTM

File: apiserver/Autoscaling.md (hunk 3 above)
Comment (blublinsky) [reply apiserver/Autoscaling.md:7]: fixed

File: apiserver/Autoscaling.md (hunk 4 above)
Comment (blublinsky) [reply apiserver/Autoscaling.md:12]: fixed

File: apiserver/Autoscaling.md (hunk 5 above)
Comment (blublinsky) [reply apiserver/Autoscaling.md:13]: fixed

**Comment by blublinsky** [comment]
@kevin85421 we are good with this one

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**